internal/types/option.go

```diff
@@ -0,0 +1,117 @@
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// Option is a more constrained subset of the code in
+//     https://github.com/moznion/go-optional
+//
+// It is not using an external go-optional lib directly because none
+// has toml unmarshal support and also because there is no support for
+// complex types in UnmarshalTOML (it only gives a single
+// reflect.Value of type any) so our optional "lib" must come with
+// limitations for this.
+//
+// Unfortunately there is no way I could find to import the code and
+// limit the supported types. So this is a copy of the subset.
+// Fortunatly the code is small, targeted and easy to follow so it
+// should not be too bad.
+
+type OptionTomlTypes interface {
+	~int | ~bool | ~string
+}
+
+// Option is a subset of github.com/moznion/go-optional for use with toml
+
+// Option is a data type that must be Some (i.e. having a value) or None (i.e. doesn't have a value).
+// This type implements database/sql/driver.Valuer and database/sql.Scanner.
+type Option[T OptionTomlTypes] []T
```

Comment:
Fwiw, we could call this `Optional` too which I actually slightly prefer but then `Option` is what rust uses and lots of people are familar with this terminology now.